### PR TITLE
Cadams/expression and grammar additions

### PR DESCRIPTION
**Pull request by chris-chen-creates**
Opening PR for expression and grammar changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,9 +27,12 @@ A simple terminal-based calculator 🧮
 ## Grammar
 
 ```
-expression      -> addition ;
-addition        -> multiplication ( "+" multiplication )* ;
-multipliciation -> negative ( "*" negative )* ;
+expression      -> subtraction ;
```
Comment (angles-n-daemons) [inline README.md:30]: 🎉

File: src/Parser.ts [hunk 2]
```diff
@@ -66,26 +97,29 @@ export class Parser {
...
+    this.consume(TokenType.RIGHT_PAREN)
+    return new Group(expr)
   }
 
   // END GRAMMAR FUNCTIONS
 
   match(ttype: TokenType): Boolean {
-    if (this.check(ttype)) {
-      return true
+    while (!this.isAtEnd()) {
```
Comment (angles-n-daemons) [inline src/Parser.ts]: why the while statement?

File: src/Parser.ts [hunk 3]
```diff
@@ -94,20 +128,31 @@ export class Parser {
   }
 
   consume(ttype: TokenType) {
-    this.advance()
+    while (!this.isAtEnd()) {
```
Comment (angles-n-daemons) [inline src/Parser.ts]: checking in on the while statement here too

File: src/__tests__/Parser.test.ts [hunk 4]
```diff
@@ -73,50 +74,63 @@ test('consume fails on incorrect token', () => {
...
+  expect(parseTree).toEqual(
+    new Addition(new Num(5), new Multiplication(new Num(4), new Num(3)))
+  )
+})
+
+test('parse number test 3', () => {
+  const parseTree = parse(scan('5 + 4 * 3 + 8'))
+  expect(parseTree).toEqual(
+    new Addition(
+      new Addition(new Num(5), new Multiplication(new Num(4), new Num(3))),
```
Comment (angles-n-daemons) [inline src/__tests__/Parser.test.ts:121]: nice!

File: src/__tests__/Parser.test.ts [hunk 5]
```diff
@@ -126,23 +140,30 @@ test('parse nested group properly', () => {
...
 
-
 test('parse throws an error if there are unparsed tokens', () => {
   expect(() => {
-    parse(scan('(3 + 4) 4'))
+    parse(scan('(3 + 4) 5'))
   }).toThrow(/.*unparsed tokens.*/)
 })
+
+test('is addition also valid multiplaction?', () => {
```
Comment (angles-n-daemons) [inline src/__tests__/Parser.test.ts:158]: this should throw an error - can you instead wrap this in an expect.toThrow? ie:

```ts
expect(() => {
   let parser = new Parser(scan('5 + 4'))
   parser.multiplication()
}).toThrow(...)

```

File: src/__tests__/Parser.test.ts [hunk 6]
```diff
@@ -126,23 +140,30 @@ test('parse nested group properly', () => {
...
+  let parser = new Parser(scan('5 + 4'))
+  console.log(parser.multiplication())
+})
+// ^^^ NO
```
Comment (angles-n-daemons) [inline src/__tests__/Parser.test.ts]: let's remove these comments, i think if read the tests should explain themselves

**Review by angles-n-daemons — CHANGES_REQUESTED** [review]
Looks good 🙌 , had a couple questions and requests - feel free to reach out if you have any questions

File: src/Parser.ts (hunk 2 above)
Comment (chris-chen-creates) [reply src/Parser.ts]: I think I wrote this back when I had a less of an understanding of the difference between if and whiles. I can see now that there is no reason for it to do a looping check to see if it's at the end within this code.

File: src/Parser.ts (hunk 3 above)
Comment (chris-chen-creates) [reply src/Parser.ts]: Reposting what I wrote on the previous comment:
"I think I wrote this back when I had a less of an understanding of the difference between if and whiles. I can see now that there is no reason for it to do a looping check to see if it's at the end within this code."

File: src/__tests__/Parser.test.ts (hunk 6 above)
Comment (chris-chen-creates) [reply src/__tests__/Parser.test.ts]: Got those comments removed

**Review by angles-n-daemons — APPROVED** [review]
Nice! One more to go!

**Pull request merged**